src/ktopt_planning_context.cpp

```diff
@@ -48,6 +48,50 @@ void KTOptPlanningContext::solve(planning_interface::MotionPlanResponse& res)
   const auto joint_model_group = getPlanningScene()->getRobotModel()->getJointModelGroup(getGroupName());
   RCLCPP_INFO_STREAM(getLogger(), "Planning for group: " << getGroupName());
 
+  // extract position and velocity bounds
+  std::vector<double> lower_position_bounds;
+  std::vector<double> upper_position_bounds;
+  std::vector<double> lower_velocity_bounds;
+  std::vector<double> upper_velocity_bounds;
+  for (const auto& joint_model : joint_model_group->getActiveJointModels())
+  {
+    const std::string& joint_name = joint_model->getName();
+    const moveit::core::VariableBounds& bounds = joint_model->getVariableBounds()[0];
+
+    lower_position_bounds.push_back(bounds.min_position_);
+    upper_position_bounds.push_back(bounds.max_position_);
+    lower_velocity_bounds.push_back(-bounds.max_velocity_);
+    upper_velocity_bounds.push_back(bounds.max_velocity_);
+
+    // std::cout << "Joint " << joint_name << ": Position ["
+    //           << bounds.min_position_ << ", " << bounds.max_position_
+    //           << "], Velocity [-" << bounds.max_velocity_ << ", "
+    //           << bounds.max_velocity_ << "]" << std::endl;
+  }
+  int num_positions = plant.num_positions();
+  int num_velocities = plant.num_velocities();
+
+  // Ensure the bounds have the correct size
+  if (lower_position_bounds.size() != num_positions || upper_position_bounds.size() != num_positions)
```

Comment:
you might get compiler warnings in comparing `int` with `size_t`, which requires a `static_cast` to resolve